java/ql/src/Likely Bugs/Likely Typos/SelfAssignment.ql

```diff
@@ -23,7 +23,9 @@ predicate toCompare(VarAccess left, VarAccess right) {
 }
 
 predicate local(RefType enclosingType, VarAccess v) {
-  enclosingType = v.getQualifier().(ThisAccess).getType()
+  exists(ThisAccess thisAccess | thisAccess = v.getQualifier() |
+    enclosingType = thisAccess.getType() and thisAccess.isOwnInstanceAccess()
+  )
```

Comment:
This change is not quite right, the code also attempts to handle the case when two variables refer to the same field in an outer scope. The code here could be improved to handle more weird corner-cases, but that's outside the scope of this PR, so let's just revert this change.